pkg/commands/resolver.go

```diff
@@ -85,9 +84,13 @@ func gobuildOptions(bo *options.BuildOptions) ([]build.Option, error) {
 		}
 	}
 
+	if multiplatform := bo.Platform == "all" || strings.Contains(bo.Platform, ","); multiplatform && bo.OSVersion != "" {
+		return nil, fmt.Errorf("cannot use --osversion with --platform=all or multiple platforms")
```

Comment:
:/

Reply:
Yeah, I don't love it. If we wanted to support this I think we should do it in `--platform` somehow, but I don't have a great idea for how to express that. I don't know if anybody has a need for it right now, I only need it to build a single platform+osversion.